Make athena resources optional, remove references to AWS-ASDI, and some minor changes

emileten opened the pull request:
### Major changes 
- make `athena` resources optional through a new `config` parameter named `athena_table` (boolean). See https://github.com/NASA-IMPACT/active-maap-sprint/issues/696#issuecomment-1698062273.

### Minor changes
- add `moto` to extra libs (for nisar-sim -- we can remove this once I created manual mocks but it takes a little bit of time, so adding this dependency for now)
- use linux/amd64 platform in all docker images (avoiding bugs I sometimes encounter when building images with my new macbook)
- fix diagram drawing (was all dark), rename to stactools instead of AWS-ASDI
- use PROJECT for tags, instead of hard coded AWS-ASDI
- add a STAGE and add project, stage and pipeline to a `deployment` `pydantic-settings` `BaseSettings`  model. 
- fix a couple bugs in the `Pipeline` pydantic model

emileten commented:
@sbquinlan do you think you have some time to look at this ? Heard you're working on ASDI -- and this repo was initially created for ASDI. 

The purpose of this PR was to make the infra a little bit more flexible to serve MAAP's ingestions needs, so that we don't maintain our own 'ingestion pipeline' infra/code and instead share the same definitions.

Thanks !

sbquinlan commented:
I'd probably just stamp it. The code looks good and I haven't deployed this. Do you want @sharkinsspatial's review instead?

sharkinsspatial commented:
@emileten I have some comments on this.  I’ll get the review up for you today.

sharkinsspatial commented on setup.py:
```diff
@@ -10,20 +10,15 @@
-    "pytest-cov",
-    "pytest",
-]
+extras_require_test = ["flake8", "black", "pytest-cov", "pytest", "moto"]
```
@emileten As we discussed a while back, I try to avoid `moto` in most of my libraries for 2 reasons.
1. It sometimes has the the effect of hiding service integration points and causes people to reduce separation of concerns in their code.  If I'm forced to explicitly mock and AWS service integration point when I write my code then it often leads to cleaner code where these service calls get encapsulated.
2. Explicit, limited surface area mocks make it much easier for reviewers to reason about what exactly the code is doing.

sharkinsspatial replied:
Ideally we should just go ahead and write the mocks.  Most of the mocking you'll need is already available in https://github.com/developmentseed/stactools-pipelines/blob/main/stactools_pipelines/pipelines/conftest.py.  But I do think we need to improve the documentation for pipeline creators to show how to use these beyond just saying "Look at the example Sentinel-1 pipeline" so we should definitely explain this a bit better.

emileten replied:
I reverted this.

>  Ideally we should just go ahead and write the mocks 

Like I said this is really time consuming. It will be up to the user to write or not write these tests for his/her pipeline.

sharkinsspatial commented on README.md:
This PR is mixing several concerns 
1. Some renaming
2. Making the Athena resource optional
3. A bunch of small changes

Mixing concerns makes PRs more difficult to review and also makes the project's commit history harder to reason about.  For future PRs let's try to reduce the scope of the PR to only the relevant, necessary changes.

emileten replied:
Agreed. Like said below, I'll keep that in mind for the next times but will keep this PR nevertheless to get things done (I reverted a couple non essential changes).

sharkinsspatial commented on image_builder.py:
```diff
@@ -21,6 +21,7 @@ def build_and_push(dockerfile: str, tag: str, pipeline_id: str):
         buildargs={
             "pipeline": pipeline_id,
         },
+        platform="linux/amd64",
```
Can you elaborate on the rationale for this change?

emileten replied:
I couldn't build on my M2 macbook without this.

That said it's not a great solution to add this arbitrarily everywhere. I reverted this, will add manually in the meanwhile.

sharkinsspatial commented on stactools_pipelines/cdk/app.py:
```diff
@@ -1,16 +1,17 @@
 from stactools_pipelines.models.pipeline import Pipeline
+from stactools_pipelines.models.deployment import Deployment
+
+deployment = Deployment()
```
@emileten As mentioned in the initial comment I'd ideally like to see these `deployment` logic changes in a separate PR so we don't have a lot of unrelated commits intermingled.

emileten replied:
Agreed, I'll keep that in mind for the next times. If that's okay, for this time I'll keep this change here since it's not large.

sharkinsspatial reviewed (changes requested):
This PR is mixing several concerns 
1. Some renaming
2. Making the Athena resource optional
3. A bunch of small changes

Mixing concerns makes PRs more difficult to review and also makes the project's commit history harder to reason about.  For future PRs let's try to reduce the scope of the PR to only the relevant, necessary changes.

emileten commented:
Thanks for the review @sharkinsspatial, I'll open a new PR with only the necessary changes I had made to make it work for MAAP, and have the other changes in a separate branch.

emileten commented on stactools_pipelines/cdk/app.py:
```diff
@@ -1,16 +1,17 @@
-# Required environment variables
-pipeline = os.environ["PIPELINE"]
-stack_name = pipeline.replace("_", "-")
+stack_name = (
```
Done https://github.com/developmentseed/stactools-pipelines/pull/42/commits/dc0d91da0c7cbbb40bd15622b33b68728cdcfdec

emileten commented on stactools_pipelines/models/pipeline.py:
```diff
@@ -13,22 +13,31 @@ class Pipeline(BaseModel):
+    inventory_location: Optional[str] = None
+    historic_frequency: Optional[int] = None
+    initial_chunk: Optional[str] = None
+    athena_table: Optional[bool] = False
```
Done

emileten commented:
@sharkinsspatial Can we merge ? I addressed your comments (ignore what I said above, I am sticking with this PR). 


Also, a broader comment regarding your statement in the comments above : 

> which [non standard inventories] as we've discussed previously, stactools-pipelines was not really intended for

I kind of get from this statement that you are reluctant to this change, am I correct? If so, I would like to stress the purpose of this PR : stactools-pipelines is an AWS tool to orchestrate parallel STAC ingestions from S3 objects. This is a task that we have to do in other projects than ASDI. Therefore, it made sense to me to put efforts to reuse this repository, instead of having one for each project. However, not all projects have control over what happens in the source buckets, and in particular, whether there is an AWS inventory or not. Allowing for something more flexible, like a csv inventory or whatever other inventory format, makes this repository more reusable. 

Does this make sense to you ? I think it's important to agree on this since there would be potentially more changes like those of this PR in the future, if we pursue that path. If you don't think this repo is the place for this, then we maybe we shouldn't proceed with this PR.

sharkinsspatial commented:
@emileten We're doing some cleanup on this repo to prepare for some workshops we'll be doing with AWS. In the case of non-standard inventory structures I think that the best approach might be using `stactools-pipelines` to create your basic processing infrastructure (SQS and the Lambdas) but use some other codebase to interpret your non-standard inventory and feed the appropriate messages to the SQS so that the STAC metadata can be created and ingested.  

This way we don't have to complicate `stactools-pipelines` logic too much for edge cases, but you can still use it to build out most of your processing pipeline.  In this way, anyone with any non-standard inventory can control how they use it to feed the `stactools-pipelines` queue.  I'm going to close this PR in favor of this mixed approach.
